File: core/src/main/java/kr/scripton/business/core/auth/FindUserAuthorityService.java
package kr.scripton.business.core.auth;

import org.springframework.security.core.GrantedAuthority;

import java.util.Collection;

public interface FindUserAuthorityService<USER_ID> {
	/**
	 * SSO에 연동된 서비스 중 사용자에게 허용된 서비스 도메인
	 * 추가2차 : 이번에 요청이 들어온 서비스
	 *
	 * @param userId
	 * @return
	 */
	Collection<String> findUserComponent(USER_ID userId);

	Collection<GrantedAuthority> findGrantedAuthority(String... roles);

	Collection<String> findUserRole(USER_ID userId);

	Collection<String> findUserPrincipal(USER_ID userId);

	Collection<String> findPrincipalByRoleName(String roleName);
}

File: direct/src/main/kotlin/kr/scripton/biz/direct/auth/FindUserAuthorityService.kt
package kr.scripton.biz.direct.auth

import org.springframework.security.core.GrantedAuthority

/**
 * @author archmagece
 * @since 2018-03-24 오전 12:43
 * @at spring-business-wrapper
 */
/**
 * @author archmagece
 * @since 2017-09-27
 */
interface SBFindUserAuthorityService {

	/**
	 * SSO에 연동된 서비스 중 사용자에게 허용된 서비스 도메인
	 * 추가2차 : 이번에 요청이 들어온 서비스
	 *
	 * @param userId
	 * @return
	 */
	fun findUserComponent(userId: Long?): Collection<String>

	fun findGrantedAuthority(vararg roles: String): Collection<GrantedAuthority>

	fun findUserRole(userId: Long?): Collection<String>

	fun findUserPrincipal(userId: Long?): Collection<String>

	fun findPrincipalByRoleName(roleName: String): Collection<String>

}
